### PR TITLE
Cleaning up preferences

### PR DESCRIPTION
**Pull request by robertbastian**
In preparation of moving into `icu_locale_core` crate. There's a lot missing.

https://github.com/unicode-org/icu4x/issues/5179

### CODE REVIEW

File: utils/preferences/src/extensions/unicode/keywords/calendar.rs [hunk 1]
```diff
@@ -4,34 +4,38 @@
...
-    "coptic" => Coptic,
-    "dangi" => Dangi,
-    "ethioaa" => Ethioaa,
-    "ethiopic" => Ethiopic,
-    "gregory" => Gregory,
-    "hebrew" => Hebrew,
-    "indian" => Indian,
-    "islamic" => Islamic(IslamicCalendarAlgorithm) {
+enum_keyword!(
+    /// TODO
```
Comment (younies) [inline utils/preferences/src/extensions/unicode/keywords/calendar.rs:11]: todo what ?

File: utils/preferences/src/extensions/unicode/keywords/calendar.rs [hunk 2]
```diff
@@ -4,34 +4,38 @@
...
         "rgsa" => Rgsa
-    },
-    "iso8601" => Iso8601,
-    "japanese" => Japanese,
-    "persian" => Persian,
-    "roc" => Roc
+});
+
+enum_keyword!(
+    /// TODO
```
Comment (younies) [inline utils/preferences/src/extensions/unicode/keywords/calendar.rs:20]: ditto

File: utils/preferences/src/extensions/unicode/keywords/currency_format.rs [hunk 3]
```diff
@@ -2,7 +2,11 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-crate::enum_keyword!(CurrencyFormatStyle {
-    "standard" => Standard,
-    "account" => Account
+use crate::extensions::unicode::enum_keyword;
+
+enum_keyword!(
+    /// TODO
```
Comment (younies) [inline utils/preferences/src/extensions/unicode/keywords/currency_format.rs:8]: ditto and the others.

**Review by younies — DISMISSED** [review]
Fix CI

File: utils/preferences/src/extensions/unicode/keywords/calendar.rs (hunk 1 above)
Comment (robertbastian) [reply utils/preferences/src/extensions/unicode/keywords/calendar.rs:11]: todo fill this in

**Review by younies — APPROVED** [review]

**Pull request merged**